### PR TITLE
Blind 75 PR 1

### PR DESCRIPTION
**Pull request by antonarboleda**
Here are a few algos that I've worked on over the past two days. 

My main takeaways overall are
- I need to improve translating ideas to code. Forcing myself to challenge/and or validate my assumptions through practice will be my best option
- I'm getting closer. It's around 40 min in practice to solve mediums. I'd like to get it down closer to 30 for mediums. 
- I still don't have Mediums mastered. I'd like to get to a point where I can do 9/10 flawlessly in 35 minutes or less.

### CODE REVIEW

File: blind_75/reorder_list.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# https://leetcode.com/problems/reorder-list/
```
Comment (antonarboleda) [inline blind_75/reorder_list.py:1]: Note: This algo doesn't work and compile. I spent 50 minutes or so but got stuck and started doing linked list problems

File: blind_75/median_finder.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# https://leetcode.com/problems/find-median-from-data-stream/
```
Comment (antonarboleda) [inline blind_75/median_finder.py:1]: @marktai

File: blind_75/encode_and_decode_strings.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+# Machine 1:
+# Codec encoder = new Codec();
+# String msg = encoder.encode(strs);
+# Machine 1 ---msg---> Machine 2
+
+# Input: dummy_input = [""]
+# Output: [""]
+
+"""
+-------- High Level Idea/Algorithm
```
Comment (antonarboleda) [inline blind_75/encode_and_decode_strings.py:40]: @marktai

File: blind_75/reorder_list.py (hunk 1 above)
Comment (antonarboleda) [reply blind_75/reorder_list.py:1]: I also didn't do pseudocode here which is probably why I struggled so much

File: blind_75/reorder_list.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    # l1 is longer
+    dummy_head = cur = Node(-1)
+    while l1 and l2:
+        tmp_l1 = l1.next
+        tmp_l2 = l2.next
+        cur.next = l1
+        cur = cur.next
+        cur.next = l2
+        l1 = tmp_l1
+        l2 = tmp_l2
```
Comment (antonarboleda) [inline blind_75/reorder_list.py:52]: Way too much swapping here. Could have just done 

```
            if l1.val < l2.val:
                cur.next = list1
                l1 = l1.next
            else:
                cur.next = l2
                l2 = l2.next
            cur = cur.next
```

File: blind_75/reorder_list.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
+# https://leetcode.com/problems/reorder-list/ 
+# problem 143 Linked List 
+# You are given the head of a singly linked-list. The list can be represented as:
+
```
Comment (antonarboleda) [inline blind_75/reorder_list.py:4]: @marktai 

This one will be hard to follow but here's a LC hard problem which is really just 3 LC easy/mediums in one. I think had I been more clear and thorough in problem solving I could have solved it

File: blind_75/encode_and_decode_strings.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        while i < len(s):
+            while s[j] != "#":
+                j += 1
+            w_length = int(s[i:j])
+            j += 1
+            word = s[j: j + w_length]
+            decoded_words.append(word)
+            j += w_length
+            i = j
+        return decoded_words
```
Comment (marktai) [inline blind_75/encode_and_decode_strings.py:104]: Oh, I actually really like this approach! I think it's very creative. 

My approach was just hex encode the string, then use commas for word separation.

File: blind_75/encode_and_decode_strings.py (hunk 3 above)
Comment (marktai) [reply blind_75/encode_and_decode_strings.py:40]: LGTM! Good test cases, though I would have added `20#` just for completion sake (I'm convinced that your code handles `20#` as word)

File: blind_75/median_finder.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+from heapq import heappush, heappop, heappushpop, heapify
+
+class MedianFinder:
+
+    def __init__(self):
+        self.smaller_numbers = [] # max heap
+        self.larger_numbers = [] # min heap
+
+    def addNum(self, num: int) -> None:
+        if not self.smaller_numbers and not self.larger_numbers:
```
Comment (marktai) [inline blind_75/median_finder.py:62]: nit: you've actually deviated from your pseudocode by deciding the smaller_numbers is the default heap.

File: blind_75/median_finder.py [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            heappush(self.smaller_numbers, -num)
+            return
+        
+        if self.larger_numbers and num > self.larger_numbers[0]:
+            heappush(self.larger_numbers, num)
+        else:
+            heappush(self.smaller_numbers, -num)
+        
+        if len(self.larger_numbers) > len(self.smaller_numbers):
```
Comment (marktai) [inline blind_75/median_finder.py:71]: teeny tiny nit: I would have made another function just called balance to really really separate out the concerns.
